### ISSUE TITLE
ESP8266 Simulation

### ISSUE THREAD

**Issue by wavesailor** [wokwi/wokwi-features#314]
Please could we get support for the ESP8266. 
I know it is older but there are still so many of them being used.

https://en.wikipedia.org/wiki/ESP8266

**Comment by urish** [comment]
Thanks for the suggestion! What would you use the ESP8266 for?

**Comment by wavesailor** [comment]
Firstly I'd like to try build all those weather stations with the ESP8266. 
This website has this one: https://randomnerdtutorials.com/esp8266-dht11dht22-temperature-and-humidity-web-server-with-arduino-ide/ 
I then like to add a battery to it if possible.

**Comment by urish** [comment]
Thanks! I opened this request [for voting](https://wokwi.com/features?sort=newest#feature-1147210537)

**Comment by Jacopo1891** [comment]
Adding ESP8266 would be good for me too. It's very cheap and easy to use. I am also using this mcu.

**Comment by juliocr1** [comment]
The use of ESP8266 is very vast, we have many projects distributed, this simulator would be a very important resource.

**Comment by urish** [comment]
> we have many projects distributed

Who is "we"? What projects do you have?

**Comment by juliocr1** [comment]
> > we have many projects distributed
> 
> Who is "we"? What projects do you have?

when I say we, I'm referring to all the electronics and microcontroller enthusiasts that we can find distributed on the internet. Regarding projects, there are a lot of projects using ESP8266, mainly IOT projects, where we can control devices and obtain telemetry. these control and data acquisition projects with low cost devices are my favorites. See this example of a module communicating with another module using a database ( https://wokwi.com/projects/341622429773202004 and https://wokwi.com/projects/341529357688242770)

**Comment by urish** [comment]
Thanks for sharing your thoughts @juliocr1!

**Comment by donald2407** [comment]
**Please get support for the ESP8266.**

**Comment by The3ven** [comment]
I'm upvoting for the esp8266 board too because I need to learn how this thing works for my FTP server

**Comment by mlrlino** [comment]
I would also like a ESP8266 with arduino uno sketch

**Comment by Ricardo-SS** [comment]
Adding ESP8266 would be good for me too. It's very cheap and easy to use. I am also using this mcu.

**Comment by infrafast** [comment]
I would like to have the simulator as well.

**Comment by urish** [comment]
Update: we open sourced a basic ESP8266 simulation library

https://github.com/wokwi/esp8266js

**Comment by hanpal** [comment]
> Thanks for the suggestion! What would you use the ESP8266 for?

I use it for all my projects since it's much cheaper than the ESP32. I have programmed about 50 ESP8266 using Wemos D1 mini or pro for home automation projects and have never even considered using ESP32 since the much cheaper ESP8266 has always been more than sufficient. And it has NOT been replaced by the ESP32.

Some examples of my projects:
- Sensors for temperature and humidity, have one in each room at home
- Sensors for light intensity, for home automation based on the light intensity in a room
- Sensors for detection of moisture on the floor in the basement
- Fan cotrol
- Control panels with buttons and displays
- Garage port/motor control
- Postbox sensors for detecting new mail
- Burglar alarm
- Motion/presence detetctors
- Intefaces to meters for electricity, water and heating
...

**Comment by Phetdanai** [comment]
Thanks to @urish for that ".zip" file.
Download Links:
[esp8266](https://github.com/wokwi/wokwi-features/files/13748628/esp8266js-main.zip)

**Comment by Laurent6012** [comment]
Hello,
And to be honest, like my father said, it's not usefull to use a hammer to kill a fly.
ESP8266 have alreday access to Wifi, enough IO for basical current and simply tasks.
**the IOT's world isn't only for rich people, not be ?**
So, using an ESP32 to manage only 1 sonde of temperature and humidity and even sending the result, my god, it's the perfect way to really becoming poor soon, i laugh.
No matter, have a good day

**Comment by Vaelek** [comment]
Man I was so psyched when I accidentally found Wowki, and then so bummed when I realized there was no 8266. As this has been open for over 2 years now, will this ever be added? I saw the post about the open source library, but I'm not really sure if or how it fits into Wowki.

**Comment by urish** [comment]
TL;DR - not very likely, unless some big commercial customer will express interest and cover the development costs.

Also, note that the ESP8266 may no longer be available anymore [in 18 months from now](https://www.espressif.com/en/products/longevity-commitment).

**Comment by Ricardo-SS** [comment]
Considering the prices prevailing on the market today and the availability of several esp32 models, I think it will no longer be an advantage to develop for the esp8266, I am migrating all my projects to the esp32 as the prices are almost the same. I think it's a waste of resources to use an esp32 in some projects, but as they are almost the same value, it's better to have the resources that are missing.

**Comment by gc2ceo** [comment]
I'm seconding ESP8266 simulation since I got some ESP8266 boards in real life.

**Comment by FelixHo97** [comment]
> Update: we open sourced a basic ESP8266 simulation library
> 
> https://github.com/wokwi/esp8266js

Hi urish, Can this version be directly connected to WiFi? Does this open source project have a corresponding instance? For example, lighting up LED lights

**Comment by urish** [comment]
Nope, there is no simulation model for the ESP8266 WIFI. Building one would require reverse engineering the WiFi stack first, similar to what I have done with the ESP32 a few years ago: https://www.youtube.com/watch?v=XmaT8bMssyQ
